### ISSUE TITLE
Can't write bool that's nulled by concatenation

### ISSUE THREAD

**Issue by syelman** [scverse/anndata#979]
Hi,
I tried to write an adata which is a result of outer concat and I get the following error.
```
TypeError: Can't implicitly convert non-string objects to strings

Above error raised while writing key 'bool' of  to /
```

Reproducible code snippet:
```python
from anndata.tests.helpers import gen_adata
import pandas as pd
import anndata as ad


a = gen_adata((5, 5), obsm_types=(pd.DataFrame,))
b = gen_adata((3, 5), obsm_types=(pd.DataFrame,))

del b.obsm["df"]
    
# del a.obsm["df"]["bool"] # this works fine

adatas = ad.concat({"a": a, "b": b}, join="outer", label="group")

adatas.write("test.h5ad")
```

Is this a bug or should we not write nulled bools like this?

**Comment by ivirshup** [comment]
I think we should create a pandas boolean column during creation, so that we can handle the null values.

**Comment by syelman** [comment]
Solved with #992
